exporters/prometheus/exporter.go

```diff
@@ -181,6 +183,16 @@ func (c *collector) Collect(ch chan<- prometheus.Metric) {
 		ch <- c.targetInfo
 	}
 
+	var resourceAttrs []attribute.KeyValue
+	if c.resourceAttributesFilter != nil {
+		for _, kv := range metrics.Resource.Attributes() {
+			filter := *c.resourceAttributesFilter
+			if filter(kv) {
+				resourceAttrs = append(resourceAttrs, kv)
```

Comment:
can we do the sanitization here (maybe even use getAttrs?), so we don't have to do it on each metric?

Reply:
I tried using getAttrs here, its a bit awkward as im also having to add a couple of params for resource keys/values that are set to nothing, but maybe that's ok